### PR TITLE
interface/modem-manager: add support for MBIM/QMI proxy clients

### PR DESCRIPTION
**Pull request by JerryLeeTaipei**
The modem's app which wants to send MBIM messages will
connect to the mbim-proxy daemon via the unix socket.

The mbim-proxy daemon uses the the abstract socket path
"mbim-proxy". The modem-manager interface needs a
corresponding update. This provides that update.

References:
https://bugs.launchpad.net/austin/+bug/1936374

### CODE REVIEW

File: interfaces/builtin/modem_manager.go
Comment (mardy) [inline interfaces/builtin/modem_manager.go:207]: Are you sure these are needed here too? It seems to me that the `modemManagerConnectedPlugAppArmorClassic` snippet is used *in addition* to the `modemManagerConnectedPlugAppArmor` snippet, so only the latter should be needed.

Comment (JerryLeeTaipei) [reply interfaces/builtin/modem_manager.go:207]: Let me test it later, thanks.

Comment (bboozzoo) [inline interfaces/builtin/modem_manager.go:207]: Shouldn't the classic variant have `peer=(label=unconfined,addr=...)` if we are to allow talking to modem manager provided by the classic host or maybe that's too strict?

**Comment by bboozzoo** [comment]
@pedronis tagged this as 'needs security review', please drop the label if there's no need.

File: interfaces/builtin/modem_manager.go
Comment (JerryLeeTaipei) [reply interfaces/builtin/modem_manager.go:207]: Yes, the part in **modemManagerConnectedPlugAppArmorClassic** is NOT needed.

File: interfaces/builtin/modem_manager.go
Comment (alexmurray) [inline interfaces/builtin/modem_manager.go:178]: This adds both the mbim-proxy and qmi-proxy addresses - can the comment please be updated to also mention qmi-proxy for completeness? Perhaps even something along the lines of:

```
# allow communicating with the mbim and qmi proxy servers, which provide
# support for talking to WWAN modems and devices which speak the Mobile 
# Interface Broadband Model (MBIM) and Qualcomm MSM Interface (QMI) 
# protocols respectively
```

So that it is clear what and why this access is granted?

File: interfaces/builtin/modem_manager.go
Comment (alexmurray) [reply interfaces/builtin/modem_manager.go]: I agree it would be good to try and scope this for the classic case but I think overall that would complicate this interface too much to try and specify an extra `label=unconfined` for just classic. Also I don't see much risk in *not* specifying this in the classic case since what is the intended threat model? Having one snap create a mbim/qmi-proxy socket and hence this rule would then allow another snap which plugs `modem-manager` to talk to that snap? I don't see that as a huge risk, plus I wonder if by specifying `unconfined` that would stop a snap from talking to an intentionally confined `mbim-proxy` etc (ie. if someone were to write an AppArmor profile for `mbim-proxy` or we decided to ship one in the future). As such I don't think this is needed but it is a good suggestion.

**Review by alexmurray — APPROVED** [review]
LGTM assuming the comment is updated so we provide a bit more details on what exactly this is required for.

File: interfaces/builtin/modem_manager.go
Comment (JerryLeeTaipei) [reply interfaces/builtin/modem_manager.go:181]: The comment was modified to be clearer.

**Review by mardy — APPROVED** [review]
Thanks!

**Comment by mvo5** [comment]
The CLA check is a bit unhappy right now - it's probably best to squash/recommit this under a "real" email instead of the users.noreply.github.com. Alternatively I could do this and commit with --author  and repush. Or are there alternative ideas?

**Comment by mvo5** [comment]
Merged via #10553

**Pull request closed without merging**
